### PR TITLE
Add FOSSA scan

### PR DESCRIPTION
**Pull request by dhaiducek**
- https://github.com/open-cluster-management/backlog/issues/18460

### CODE REVIEW

File: .github/workflows/fossa.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    name: License and Security Scan
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+
+      - name: Run FOSSA Scan
+        uses: fossas/fossa-action@v1
+        with:
+          api-key: ${{ secrets.FOSSA_API_KEY }}
```
Comment (mprahl) [inline .github/workflows/fossa.yml:21]: Could you please document how this gets populated and also what the FOSSA scan is for in general?

File: .github/workflows/fossa.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: FOSSA
+
+on:
```
Comment (mprahl) [inline .github/workflows/fossa.yml:3]: Is there a reason to not also run this on PRs? It'd be ideal to catch these issues before the PR gets merged.

Comment (dhaiducek) [reply .github/workflows/fossa.yml:3]: I agree, but the scan requires an API key, which is stored in a secret, which is not accessible to forks, so this was the next best thing.

File: .github/workflows/fossa.yml (hunk 1 above)
Comment (dhaiducek) [reply .github/workflows/fossa.yml:21]: It's a CNCF requirement, so I'll have to do a bit of homework for that. 🙂 Where should this be documented? Like in comments in the workflow? Or maybe in the contributing guide?

File: .github/workflows/fossa.yml (hunk 2 above)
Comment (mprahl) [reply .github/workflows/fossa.yml:3]: Ah, that makes sense.

File: .github/workflows/fossa.yml (hunk 1 above)
Comment (mprahl) [reply .github/workflows/fossa.yml:21]: The contributing guide would be good!

Comment (dhaiducek) [reply .github/workflows/fossa.yml:21]: Description added--let me know what you think, and I'll update the other contributing guides also.

**Review by mprahl — APPROVED** [review]
This looks great!

**Pull request merged**